firmware/arduino/chessboard/motorDriver.ino

```diff
@@ -50,33 +80,45 @@ void moveDiagonal(int dirX, int dirY, int spacesX, int spacesY)
   digitalWrite(xMotor[1], dirX);
   digitalWrite(yMotor[1], dirY);
   
-  if (numStepsX = numStepsY) 
+  if (numStepsX == numStepsY) 
   {
     for(i = 0; i < numStepsX; i++)
     {
-    digitalWrite(xMotor[0], LOW);
-    digitalWrite(yMotor[0], LOW);
-    delay(1);
-    digitalWrite(xMotor[0], HIGH);
-    digitalWrite(yMotor[0], HIGH);
+      digitalWrite(xMotor[0], LOW);
+      digitalWrite(yMotor[0], LOW);
+      delay(1);
+      digitalWrite(xMotor[0], HIGH);
+      digitalWrite(yMotor[0], HIGH);
     }
   } 
 
-  else 
+  else if (numStepsY > numStepsX && (numStepsY / numStepsX) == 2)
   {
-    float dist = sqrt(numStepsX * numStepsX + numStepsY * numStepsY);
-    float delayX = dist / numStepsX;
-    float delayY = dist / numStepsY;
+    //can I call setScale here even thought it's declared after?
+    //This should be a slope of 2
+    setScale(xMotor, 2);
 
-    for(i=0; i < dist; i++)
+    for(i = 0; i < numStepsX; i++)
+    {
       digitalWrite(xMotor[0], LOW);
-      //do i have to convert this into an int, can i keep it as a variable?
-      delay(delayX);
+      digitalWrite(yMotor[0], LOW);
+      delay(1);
       digitalWrite(xMotor[0], HIGH);
+      digitalWrite(yMotor[0], HIGH);
+    }
+  }
+  else if (numStepsY < numStepsX)
+  {
+    setScale(yMotor, 2);
 
+    for(i = 0; i < numStepsX; i++)
+    {
+      digitalWrite(xMotor[0], LOW);
       digitalWrite(yMotor[0], LOW);
-      delay(delayY);
+      delay(1);
+      digitalWrite(xMotor[0], HIGH);
       digitalWrite(yMotor[0], HIGH);
+    }
   }
```

Comment:
I have one more comment about this code, but I want to wait until you push the next set of changes. It's about the repeated for loop. See if you can figure it out :)